src/Features/Core/Portable/Diagnostics/HostAnalyzerManager.cs

```diff
@@ -12,15 +15,11 @@
 namespace Microsoft.CodeAnalysis.Diagnostics
 {
     /// <summary>
-    /// This owns every information about analyzer itself 
-    /// such as <see cref="AnalyzerReference"/>, <see cref="DiagnosticAnalyzer"/> and <see cref="DiagnosticDescriptor"/>.
-    /// 
-    /// people should use this to get <see cref="DiagnosticAnalyzer"/>s or <see cref="DiagnosticDescriptor"/>s of a <see cref="AnalyzerReference"/>.
-    /// this will do appropriate de-duplication and cache for those information.
-    /// 
-    /// this should be always thread-safe.
+    /// Provides and caches information about diagnostic analyzers such as <see cref="AnalyzerReference"/>, 
+    /// <see cref="DiagnosticAnalyzer"/> instance, <see cref="DiagnosticDescriptor"/>s.
+    /// Thread-safe.
     /// </summary>
-    internal sealed partial class HostAnalyzerManager
+    internal sealed partial class DiagnosticAnalyzerInfoCache
```

Comment:
Rename file?

Reply:
I will in a follow up PR. I actually did rename the file before but git wasn't able to figure out the diff.